conans/client/tools/scm.py

```diff
@@ -62,10 +55,11 @@ def run(self, command):
         command = "%s %s" % (self.cmd_command, command)
         with chdir(self.folder) if self.folder else no_op():
             with environment_append({"LC_ALL": "en_US.UTF-8"}) if self._force_eng else no_op():
-                if not self._runner:
-                    return check_output(command).strip()
-                else:
-                    return self._runner(command)
+                with pyinstaller_bundle_env_cleaned():
```

Comment:
Adding this ``pyinstaller_bundle_env_cleaned()`` should be the only functional change (and only affecting pyinstaller) in this PR. The rest is a pure refactor.

Reply:
This could potentially be an issue for any other system call, right? I would suggest to separate it in a different PR (that closes #6161) and open a new issue to think if we want to use it for all the other system calls.

Reply:
Oh, I already checked that before doing that PR. Basically all other system calls are very simple "version" or similar calls, which is extremely unlikely to be using the wrong SSL library, as the SCM calls are doing. The only one that really needs it is this one, and I strongly prefer to introduce this thing only as needed, not before.

Reply:
My concern is about people doing a `git clone` in the `source()` function, for example, it would be the same problem (it appears in our docs). Or calls to `self.run` from within a conanfile that could be running any tool.

But yes, I agree that we can go step by step

Reply:
The `` with pyinstaller_bundle_env_cleaned():`` is already applied to all ``self.run()`` commands via the ``ConanRunner``, has been there for a while, so we are fine there. 

It is just the other runners, those used to get the version of cmake or MSBuild, or used to detect compilers, which do not include this protection yet (and I think they will probably not be necessary)